index.js

```diff
@@ -18,10 +20,11 @@ var push = [].push
  *
  * @param {EstreeNode} tree
  * @param {EstreeComment[]} [comments]
+ * @param {boolean} isEstreeNode
  */
-export function attachComments(tree, comments) {
+export function attachComments(tree, comments, isEstreeNode = false) {
   var list = (comments || []).concat().sort(compare)
-  if (list.length) walk(tree, {comments: list, index: 0})
+  if (list.length) walk(tree, {comments: list, index: 0, isEstreeNode})
```

Comment:
I don’t think it’s needed to pass this variable in state — it’s already in scope